tests/conformance/conformance.sh

```
@@ -0,0 +1,133 @@
+    	fi
+
+	CLUSTER=$(echo "${CLUSTER}" | awk '{print tolower($0)}')
+	export CLUSTER
+
+    	# randomly select region
+    	REGIONS=(us-east-1 us-east-2 us-west-1 us-west-2)
+    	export CHANGE_ID=${CHANGE_ID:-${BUILD_ID}}
+    	i=$(( CHANGE_ID % ${#REGIONS[@]} ))
+    	export TF_VAR_tectonic_aws_region="${REGIONS[$i]}"
+    	export AWS_REGION="${REGIONS[$i]}"
+    	echo "selected region: $TF_VAR_tectonic_aws_region"
+    	echo "cluster name: $CLUSTER"
+}
+
+function cleanup() {
+	docker run -i -v ${WORKSPACE}:${PROJECT} ${MNT_SECRETS} quay.io/coreos/tectonic-builder:v1.9 /bin/bash <<EOF
+      cd ${PROJECT}
+      
+      export PLATFORM=${PLATFORM}
+      cluster_check "${CLUSTER}"
+      export AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+      export AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+      export TF_VAR_tectonic_pull_secret_path=${TF_VAR_tectonic_pull_secret_path}
+      export TF_VAR_tectonic_license_path=${TF_VAR_tectonic_license_path}
+      export TF_VAR_tectonic_cluster_name=${TF_VAR_tectonic_cluster_name}
+      
+      make destroy
+EOF
+}
+trap cleanup EXIT
+
+
+export KUBECTL=${WORKSPACE}/kubectl
+curl -o ${KUBECTL} https://storage.googleapis.com/kubernetes-release/release/v1.6.3/bin/linux/amd64/kubectl && chmod +x ${KUBECTL}
+function kubectl() {
+  local i=0
+
+  echo "Executing kubectl $@"
+  while true; do
+    (( i++ )) && (( i == 100 )) && echo "kubectl failed, giving up" && exit 1
+
+    set +e
+    out=$($KUBECTL "$@" 2>&1)
+    statu=$?
+    set -e
+
+    if [[ "$out" == *"AlreadyExists"* ]]; then
+      echo "$out, skipping"
+      return
+    fi
+
+    echo "$out"
+    if [ "$statu" -eq 0 ]; then
+      return
+    fi
+
+    echo "kubectl failed, retrying in 5 seconds"
+    sleep 5
+  done
+}
+
+function wait_for_pods() {
+  set +e
+  local i=0
+  echo "Waiting for pods in namespace $1"
+  while $KUBECTL -n "$1" get po -o custom-columns=STATUS:.status.phase,NAME:.metadata.name | tail -n +2 | grep -v '^Running'; do
```

once you add `-o pipefail` this while check will not do what we want. if kubectl returns an error, then the error will not be swallowed by the pipe, even if you do `+e`. As a result, if kubectl returns any error, then the while condition will evaluate to false and the function will exit.